packages/astro/src/content/data-store.ts

```diff
@@ -371,5 +468,13 @@ function dataStoreSingleton() {
 	};
 }
 
+// TODO: find a better place to put this image
+export function contentModuleToId(specifier: string, filePath: string) {
+	const params = new URLSearchParams(specifier);
+	params.set('importer', filePath);
```

Comment:
As mentioned in the other comment: rather than using an `importer` param, could you use the filePath as the specifier? So `data/iguana.mdx?astroContentModuleFlag`. I used `improter` for images because they needed to be resolved relative to the source file, but that's not eneded here.